wgpu/src/lib.rs

```diff
@@ -3150,7 +3151,16 @@ impl<'a> RenderPass<'a> {
 
     /// Draws primitives from the active vertex buffer(s).
     ///
-    /// The active vertex buffers can be set with [`RenderPass::set_vertex_buffer`].
+    /// The active vertex buffer(s) can be set with [`RenderPass::set_vertex_buffer`].
+    /// Does not use an Index Buffer. If you need this see [`RenderPass::draw_indexed`]
+    ///
+    /// Panics if vertices Range is not within 0..BufferLen / BufferStride.
+    ///
+    /// vertices: The range of vertex's being used from the Vertex Buffer.
+    /// instances: Change the vertex data used for any instance-rate vertex attributes
+    ///     from a set instance buffer. The instance index is passed to the vertex
```

Comment:
again, this needs some love for not actually using vertex buffers. Something along the line that this is what advances the stepping on instance stepped vertex buffer if any set. Note on `0..1` is really good, newcomers are bound to trip on that otherwise.
Haven't checked what WebGPU specs here, should be able to find some nice text blocks there

Reply:
Any idea's on how this could be rewritten to be better? and yeah the 0..1 got me in the beginning too as I thought this originally was for choosing the next vertex. Since this had no documentation on what it does. 

Also I am not sure if the WebGPU specs cover all these for everything or not. by this i mean for Metal, opengl, Vulkan etc. If it does then we could probably be safe in using whatever they give. If not maybe need to phrase it to include all instances?

Reply:
oh but what you've written now sounds quite good I think and describes what it does with applicable vertex buffers :)
Tbh I haven't read everything in the spec about it, but I think it covers this well enough and there should be no backend differences.

Reply:
I changes this to be an actual usage Example of how its done as this shows Way more than a description would in most cases.